.buildkite/scripts/initCloudEnv.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+REPO_DIR=$(pwd)
+
+exportAwsSecrets() {
+  echo "~~~ Exporting AWS secrets"
+  export AWS_ACCESS_KEY_ID=$BEATS_AWS_ACCESS_KEY
+  export AWS_SECRET_ACCESS_KEY=$BEATS_AWS_SECRET_KEY
+  export TEST_TAGS="${TEST_TAGS:+$TEST_TAGS,}aws"
+
+  # AWS_REGION is not set here, since AWS region is taken from *.tf file:
+  # - x-pack/metricbeat/module/aws/terraform.tf
+  # - x-pack/filebeat/input/awscloudwatch/_meta/terraform/variables.tf
+}
+
+terraformApply() {
+  echo "~~~ Terraform Init on $MODULE_DIR"
+  terraform -chdir="$MODULE_DIR" init
+
+  TF_VAR_BRANCH=$(echo "${BUILDKITE_BRANCH}" | tr '[:upper:]' '[:lower:]' | sed 's/[^a-z0-9-]/-/g')
+  TF_VAR_CREATED_DATE=$(date +%s)
+  export TF_VAR_BUILD_ID="${BUILDKITE_BUILD_ID}"
+  export TF_VAR_ENVIRONMENT="ci"
+  export TF_VAR_REPO="${REPO}"
+  export TF_VAR_BRANCH
+  export TF_VAR_CREATED_DATE
+
+  echo "~~~ Terraform Apply on $MODULE_DIR"
+  terraform -chdir="$MODULE_DIR" apply -auto-approve
+}
+
+terraformDestroy() {
+  cd $REPO_DIR
+  find "$MODULE_DIR" -name terraform.tfstate -print0 | while IFS= read -r -d '' tfstate; do
+    cd "$(dirname "$tfstate")"
+    if ! terraform destroy -auto-approve; then
+      return 1
+    fi
+    cd -
+  done
+  return 0
+}
+
+trap 'terraformDestroy' EXIT
+exportAwsSecrets
+
+max_retries=2
+timeout=5
+retries=0
+
+while true; do
+  echo "--- Setting up Terraform"
+  out=$(terraformApply 2>&1)
```

Comment:
`terraformApply | tee out` seems shorter

Reply:
As far as I understand the following `exit_code=$?` will return exitcode of `tee` instead of `terraformApply`